web/blog/zeek-in-the-shell/index.md

```diff
@@ -0,0 +1,179 @@
+     shell "zeek -r - LogAscii::output_to_stdout=T \
+            JSONStreaming::disable_default_logs=T \
+            JSONStreaming::enable_log_rotation=F \
+            json-streaming-logs"
+     | parse zeek-json
+    suricata: >
+     shell "suricata -r /dev/stdin \
+            --set outputs.1.eve-log.filename=/dev/stdout \
+            --set logging.outputs.0.console.enabled=no"
+     | parse suricata
+```
+
+The difference stands out when you look now at the pipeline definition:
+
+```text title=Zeek
+zeek
+| where #type == "zeek.conn"
+| summarize n=count_distinct(id.orig_h) by id.resp_h
+| sort n desc
+```
+
+```text bash title=Suricata
+suricata
+| where #type == "suricata.flow"
+| summarize n=count_distinct(src_ip) by dst_ip
+| sort n desc
+```
+
+It's pretty convenient to drop packets into a Tenzir pipeline, process them with
+our favorite tools, and then perform fast in-situ analytics on them. The nice
+thing is that operators compose: a new operator automatically works with all
+existing ones.
+
+## How does it work?
+
+Under the hood, the `shell` operator forks the `tenzir` process and spawns
+`zeek` or `suricata` as child process. The operator then rewires stdin and
+stdout of the `tenzir` process such their byte streams get routed through stdin
+and stdout of the forked child.
+
+Compare this to the "standard" approach:
+
+![Piping Zeek to Tenzir](zeek-to-tenzir-pipe.excalidraw.svg)
+
+Using `shell`, we have this scenario:
+
+![Shelling out to Zeek](zeek-to-tenzir-shell.excalidraw.svg)
+
+The `shell` operator is bytes-to-bytes transformation. This means it can only
+interact with other operators that also work on bytes. In the above examples, we
+used the shell pipe to connect stdin and stdout of two processes. We can also
+use Tenzir's native pipe with the [`file`](/docs/connectors/file) loader:
+
+```
+load file trace.pcap
+| zeek
+| where 6.6.6.6
+| write json
+```
+
+Got a PCAP trace via Kafka? Just exchange the `file` loader with
+[`kafka`](/docs/connectors/kafka):
+
+```
+load kafka -t artifact
+| zeek
+| where 6.6.6.6
+| write json
+```
+
+You may not always sit in front of a command line and are able to pipe data from
+a Unix tool into a Tenzir pipeline. For example, when you use our
+[app](/setup-guides/use-the-app) or the [REST API](/rest-api). This is where the
+`shell` operator shines. The diagram above shows how `shell` shifts the entry
+point of data from a tool to the Tenzir process. You can consider `shell` your
+escape hatch to reach deeper into a specific Tenzir node, as if you had a native
+shell.
+
+:::info Unsafe operators
+Naturally, we must add that this is also a security risk and can lead to
+arbitrary shell access. This is why the `shell` operator and a few other ones
+with far-reaching side effects are disabled by default. If you are aware of the
+implications, you can remove this restriction by setting
+`tenzir.allow-unsafe-pipelines: true` in the `tenzir.yaml` of the respective
+node.
+:::
```

Comment:
They are only disabled when using a node, but I think it's fine to ignore this detail here.